saleor/graphql/product/mutations/products.py

```diff
@@ -1519,14 +1524,29 @@ def perform_mutation(cls, _root, info, **data):
                 image=image_data, alt=alt, type=ProductMediaTypes.IMAGE
             )
             create_product_thumbnails.delay(media.pk)
-        else:
-            oembed_data, media_type = get_oembed_data(media_url, "media_url")
-            media = product.media.create(
-                external_url=oembed_data["url"],
-                alt=oembed_data.get("title", alt),
-                type=media_type,
-                oembed_data=oembed_data,
-            )
+        if media_url:
+            # Remote URLs can point to the images or oembed data.
+            # In case of images, file is downloaded. Otherwise we keep only
+            # URL to remote media.
+            if is_image_url(media_url):
+                validate_image_url(media_url, "media_url", ProductErrorCode.INVALID)
+                filename = get_filename_from_url(media_url)
+                image_data = requests.get(media_url, stream=True)
```

Comment:
I am wondering about the security. In this case we don't have any validation against malware/suspicious files so we can upload anything and I am not sure that checking mime type and content type is enough.  Correct me if I am wrong.
 @maarcingebala @patrys

Reply:
Maybe we could just add validation of the mime types and support these? https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types#image_types

Reply:
Ah nevermind, `is_image_mimetype` already does it. I'm not sure if we can check anything else here.

Reply:
Can I mark it as resolved?